lib/Settings/Settings.js

```diff
@@ -62,12 +62,32 @@ class Settings extends Component {
     const navlinks = (section) => {
       const navItems = section.pages.map((p) => {
         const search = props.location.search || '';
-        const link = `${props.match.path}/${p.route}${search}`;
+        const link = `${props.match.path}/${p.route}`;
+        let href = link;
+        let isActive = false;
+
+        // NavListItem allows for isActive to be set independently, rather than
+        // strict internal comparison of the href and the `activeLink`
```

Comment:
Maybe my beef is with the API then, because this doesn't seem correct. `isActive` should be true whenever `props.location.pathname` matches `link`, and we shouldn't have to worry about it. In fact, we shouldn't be _able_ to worry about it.

Reply:
Oh @zburke, you little little man. You can hardly see the ground beneath your feet, let alone the view from 10,000'. The problem here is that `<NavListItem>` _tries_ to correctly set `isActive` based on the `activeLink` value in the context it consumes, but that value may only include a partial-path in a nested route, so sometimes we _have_ to do that worrying up here at a higher level.